scripts/suse/apt/aptnotify

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+
+import os
+import hashlib
+
+CK_PATH = "/var/cache/salt/minion/rpmdb.cookie"
+DPKG_PATH = "/var/lib/dpkg/status"
+
+def _get_mtime():
+    """
+    Get the modified time of the Package Database.
+
+    Returns:
+        Unix ticks
+    """
+    return os.path.exists(DPKG_PATH) and int(os.path.getmtime(DPKG_PATH)) or 0
+
+
+def _get_checksum():
+    """
+    Get the checksum of the Package Database.
+
+    Returns:
+        hexdigest
+    """
+    digest = hashlib.sha256()
+    with open(DPKG_PATH, "rb") as rpm_db_fh:
+        while True:
+            buff = rpm_db_fh.read(0x1000)
+            if not buff:
+                break
+            digest.update(buff)
+    return digest.hexdigest()
+
+
+def dpkg_post_invoke():
+    """
+    Hook after the package installation transaction.
+    """
+    if 'SALT_RUNNING' not in os.environ:
```

Comment:
We need to make sure that `SALT_RUNNING` is set for this to work

Reply:
Hmmm, not in this case. Here we're in the APT plugin, which always run when there is apt transaction.

In this case, we want to avoid changing the cookie if the `SALT_RUNNING` variable is set, and it's only been set by Salt aptpkg module when calling apt.

Changing the cookie will trigger the beacon event and therefore a pkg list refresh, and we don't want that when the apt execution was made by Salt.

Reply:
My comment was too easy to misunderstand. Let me try again ;)

We need to make sure that aptpkg is setting `SALT_RUNNING`, just like yumpkg and zypperpkg do. aptpkg uses the same mechanism that zypperpkg does, i.e. setting `SALT_RUNNING` can be done in the configuration file. I didn't check if it is already configured to do so, but I think not (given the bug report we had because we didn't configure it for zypperpkg). Either way, we should not forget this step.